python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -934,3 +947,75 @@ def export_records(self, refs: Iterable[DatasetIdRef]) -> Mapping[str, Datastore
                 all_records[name] = record_data
 
         return all_records
+
+    def export(
+        self,
+        refs: Iterable[DatasetRef],
+        *,
+        directory: Optional[ResourcePathExpression] = None,
+        transfer: Optional[str] = "auto",
+    ) -> Iterable[FileDataset]:
+        # Docstring inherited from Datastore.export.
+        if transfer is not None and directory is None:
+            raise TypeError(f"Cannot export using transfer mode {transfer} with no export directory given")
+
+        if transfer == "move":
+            raise TypeError("Can not export by moving files out of datastore.")
+
+        # Exporting from a chain has the potential for a dataset to be
+        # in one or more of the datastores in the chain. We only need one
+        # of them since we assume the datasets are the same in all (but
+        # the file format could be different of course since that is a
+        # per-datastore configuration).
+        # We also do not know whether any of the datastores in the chain
+        # support file export.
+
+        # Ensure we have an ordered sequence that is not an iterator or set.
+        if not isinstance(refs, Sequence):
+            refs = list(refs)
+
+        # If any of the datasets are missing entirely we need to raise early
+        # before we try to run the export. This can be a little messy but is
+        # better than exporting files from the first datastore and then finding
+        # that one is missing but is not in the second datastore either.
+        known = [datastore.knows_these(refs) for datastore in self.datastores]
+        combined_known: dict[DatasetRef, bool] = {}
+        for known_to_this in known:
+            for ref, knows_this in known_to_this.items():
+                # If we have not known it yet, store the newer value.
+                if not combined_known.get(ref):
+                    combined_known[ref] = knows_this
```

Comment:
Maybe use `collections.Counter` for `combined_known`, you can just sum 'True' values with `update()` instead of doing an inner loop (but internally it's probably the same loop).

Reply:
I'm not really sure I understand how to use `collections.Counter` here. It did make me consider this alternative though:

```python
        refs_known: set[DatasetRef] = set()
        for known_to_this in known:
            refs_known.update({ref for ref, knows_this in known_to_this.items() if knows_this})
        missing_count = len(refs) - len(refs_known)
```
which is much shorter and seems to work.